src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -72,6 +91,33 @@ class PalantirDockerPlugin implements Plugin<Project> {
                 workingDir dockerDir
                 commandLine 'docker', 'push', ext.name
             }
+
+            dockerfileZip.with {
+                from(ext.resolvedDockerfile)
+            }
+
+            // Configure docker-compose templating
+            if (ext.resolvedDockerComposeTemplate.exists()) {
+                def dockerDependencies = project.configurations.docker.resolvedConfiguration.resolvedArtifacts
+                def templateTokens = dockerDependencies.collectEntries {
+                    def version = it.moduleVersion.id
+                    [("{{${version.group}:${version.name}}}"): version.version]
+                }
+
+                generateDockerCompose.with {
+                    from(ext.resolvedDockerComposeTemplate)
+                    into(ext.resolvedDockerComposeFile.parentFile)
```

Comment:
IMO the most likely flow is to have a docker-compose.yml.template in the root project's dir and generate a docker-copmose.yml at the same level, likely .gitignore'd